journal/journal.txt

```diff
@@ -0,0 +1,59 @@
+      suggest that when using the libusb in C++, they suggest installing
+      libsub-1.0.0, but I think libusb-dev was enough to "make" pk2cmd from the
+      source.
+    - The "sudo apt-get install sdcc" installed outdated sdcc which did not
+      have the 18f6887.h header. Need to build from downloading the latest
+      binary.
+    - However, this binary is 32-bit, so I need to install some library so that
+      Ubuntu 64bit can run 32-bit binary.
+    - But after installing libraries for fixing 32-bit issue, there was
+      `GLIBCXX_3.4.19' not found error. This issue was fixed by installing
+      libstdc++6 from a non-default location: "ppa:ubuntu-toolchain-r/test".
+2015.10.29 Friday
+    - Learning how to use Pickit2. Make sure the wire connection from Pickit2
+      to the development board is correct. There are white triangles (arrow) on
+      both the Pickit2 and the board, so the wiring should match them.
+    - Successfully programmed the chip, but I don't see the LED blinking.
+      Either external power must be provided to the board, or I'm using wrong
+      port to blink the light. Currently the port C is used in the blink.c.
+2015.10.31 Saturday
+    - Trying to turn on the LED light on the 44-pin demo board. Unsuccessful.
+2015.11.15 Sunday
+    - Maybe it is some kind of configuration problem.... Still figuring it out.
+    - The path to the PIC header file is at:
+      "/usr/local/share/sdcc/non-free/include/pic14/pic16f887.h"
+2015.11.17 Tuesday
+    - My Pickit2 is not powering the MCU (The "target" LED on Pickit2 is not
+      lit.
+    - Maybe I should try to program in Windows first.
+2015.11.27 Friday
+    - MPLAB X is used to successfully blink LEDs on the 44-pin target board.
+        1. Right click the "blink" project
+        2. Click Properties
+        3. Check Pickit 2 for the hardware tool
+        4. Check xc8 compiler (install it if not found by MPLAB X)
+        5. Choose correct PIC cihp (PIC16F887 for now)
+        6. Then the source code should compile.
+    - Sometimes, you have to program second time in order to get the code run
+      in the actual hardware..
+2015.11.21 Saturday
+    - Installed MPLAB X on Ubuntu
+2015.11.23 Monday
+    - No success. Trying a combination of configuration bits.
+2015.11.27 Friday 
+    - Success with MPLAB X for blinking the LEDs. Correct configuration is needed
+2015.11.28 Saturday
+    - Failed to blink LEDs using SDCC
+2015.12.3 Thursday
+    - The potentiometer should NOT be turned all the way in the clock-wise
+      direction. Somehow it draws power so that LED does not lit up even with
+      using MPLAB X.
+    - Was able to power the target using pk2cmd.
+    - Successful using the right configuration and powering the board with
+      pk2cmd! Now LED blinks! example3 is the working code.
+2015.12.12 Saturday
+    - Testing which configuration bits are neccessary.
+2015.12.13 Sunday
+    - Refactored some code.
```

Comment:
Add how I went through in selecting one among the Application Lifecycle Management tools